### PR TITLE
FLUID-5536: Added documentation for textfield and related components

### PR DESCRIPTION
**Pull request by jobara**
https://issues.fluidproject.org/browse/FLUID-5536

Related Infusion PR: https://github.com/fluid-project/infusion/pull/823

### CODE REVIEW

File: src/documents/TextfieldAPI.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+---
+title: Textfield
+layout: default
+category: Components
+---
+
+The **Textfield** is small component for adding aria attributes and data binding to a HTML textfield. Typically this will be used in conjunction with a controller (e.g. [**Textfield Slider**](TextfieldSlider.md), [**Textfield Stepper**](TextfieldStepper.md)) to restrict the values and pair with additional input methods.
```
Comment (cindyli) [inline src/documents/TextfieldAPI.md]: "...is a small component...". Missing "a"

File: src/documents/TextfieldAPI.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        <tr>
+            <th>Method</th>
+            <td>
+                <code>fluid.textfield(container, options);</code>
+            </td>
+        </tr>
+        <tr>
+            <th>Description</th>
+            <td>
+                Instantiates the textfield component. Provides databinding between the HTML `<input>` and the components model. Also provides options for setting an `"aria-label"` and/or `"aria-labelledby"` attribute.
```
Comment (cindyli) [inline src/documents/TextfieldAPI.md]: Shall `databinding` be 2 words?

File: src/documents/TextfieldAPI.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                The <code>setModel</code> method allows for updating the model value based on an event. This is primarily used internally for binding to the input's change event.
+            </td>
+        </tr>
+        <tr>
+            <th>Parameters</th>
+            <td>
+                <dl>
+                    <dt>event</dt>
+                    <dd>
+                        An event object including a change value.
```
Comment (cindyli) [inline src/documents/TextfieldAPI.md]: Would be helpful to provide more info about what this event is (a jQuery event with a link to jquery event system documentation) and from where the function fetches data from (`event.target.value`).

File: src/documents/TextfieldSliderAPI.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
+---
+title: Textfield Slider
+layout: default
+category: Components
+---
+
+The **Textfield Slider** is user interface component for adjusting a number value using a range input. It is paired with a [Textfield](Textfield.md) for direct entry of the value.
```
Comment (cindyli) [inline src/documents/TextfieldSliderAPI.md]: "... is a user interface component...". Missing "a"?

File: src/documents/TextfieldSliderAPI.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+## Model Paths ##
+
+The following model paths can be used with [model listeners](ChangeApplierAPI.md).
+
+* `model.value`: `null` by default but can contain any numerical value provided it is within the range (inclusive).
```
Comment (cindyli) [inline src/documents/TextfieldSliderAPI.md]: Would be nice to be clear that all values listed at line 78-81 are only default values. For example:

`model.step`: The amount to increment/decrement the model value by. Primarily this is used by the slider. Default: `1.0`

File: src/documents/TextfieldSliderAPI.md [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+## Options ##
+
+### strings ###
+
+<table>
+    <tbody>
+        <tr>
+            <th>Description</th>
+            <td>
+                The <code>strings</code> option allows you to specify strings to be used by the component and is the main location for localization. In particular this is used to set an <code>"aria-label"</code>.
```
Comment (cindyli) [inline src/documents/TextfieldSliderAPI.md]: Might be useful to indicate `aria-label` is set onto the slider. Same comment applies to other spots where aria attribute is set, for both text field slider and stepper.

File: src/documents/TextfieldStepperAPI.md [hunk 7]
```diff
@@ -0,0 +1,340 @@
+---
+title: Textfield Stepper
+layout: default
+category: Components
+---
+
+The **Textfield Stepper** is user interface component for adjusting a number value using a [Textfield](Textfield.md) for direct entry of the value, along buttons for incrementing and decrementing the value. A user can also increment/decrement the values using the up/down arrow keys.
```
Comment (cindyli) [inline src/documents/TextfieldStepperAPI.md]: "... is a user interface component...". Missing "a"?

File: src/documents/TextfieldStepperAPI.md [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+        <tr>
+            <th>Method</th>
+            <td>
+                <code>fluid.textfieldStepper(container, options);</code>
+            </td>
+        </tr>
+        <tr>
+            <th>Description</th>
+            <td>
+                Instantiates the Textfield Stepper component. Connects a a textfield and a pair of increment/decrement buttons to set a model value. Entries are restricted to numbers within a given range.
```
Comment (cindyli) [inline src/documents/TextfieldStepperAPI.md]: "... set a value...". remove "model"? Same comment applies to docs for text field and text field slider.

File: src/documents/TextfieldStepperAPI.md [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+        <tr>
+            <th>Method</th>
+            <td>
+                <code>textfieldStepper.addFocus();</code>
+            </td>
+        </tr>
+        <tr>
+            <th>Description</th>
+            <td>
+                The <code>addFocus</code> method adds the focus style classes to the <code>focusContainer</code>.
```
Comment (cindyli) [inline src/documents/TextfieldStepperAPI.md]: Add a link or explain how to define `focusContainer`.

File: src/documents/TextfieldStepperAPI.md [hunk 10]
```diff
@@ -0,0 +1,340 @@
...
+            <td>none</td>
+        </tr>
+    </tbody>
+</table>
+
+## Model Paths ##
+
+The following model paths can be used with [model listeners](ChangeApplierAPI.md).
+
+* `model.value`: `null` by default but can contain any numerical value provided it is within the range (inclusive).
```
Comment (cindyli) [inline src/documents/TextfieldStepperAPI.md]: Same comment to make clear the listed values are default values.

File: src/documents/TextfieldStepperAPI.md (hunk 9 above)
Comment (jobara) [inline src/documents/TextfieldStepperAPI.md]: Didn't add a link, but added more information to indicate that `focusContainer` is a selector.

**Comment by jobara** [comment]
@cindyli ready for more review.

File: src/documents/TextfieldSliderAPI.md [hunk 11]
```diff
@@ -0,0 +1,277 @@
...
+
+## Model Paths ##
+
+The following model paths can be used with [model listeners](ChangeApplierAPI.md).
+
+<table>
+    <thead>
+        <tr>
+            <td>Model Path</td>
+            <td>Default</td>
```
Comment (cindyli) [inline src/documents/TextfieldSliderAPI.md]: Can you use the same table format as other API docs? i was referring to [Switch API](http://docs.fluidproject.org/infusion/development/SwitchAPI.html#general-options). Also have "Default" as the last column. The same comment applies to "TextfieldStepper.md".

File: src/documents/TextfieldAPI.md [hunk 12]
```diff
@@ -0,0 +1,168 @@
+---
+title: Textfield
+layout: default
+category: Components
+---
+
+The **Textfield** is a small component for adding aria attributes and data binding to a HTML textfield. Typically this will be used in conjunction with a controller (e.g. [**Textfield Slider**](TextfieldSlider.md), [**Textfield Stepper**](TextfieldStepper.md)) to restrict the values and pair with additional input methods.
```
Comment (cindyli) [inline src/documents/TextfieldAPI.md]: Filenames for TextfieldSlider should be `TextfieldSliderAPI.md`. Same applies to TextfieldStepper.

File: src/documents/TextfieldStepperAPI.md [hunk 13]
```diff
@@ -0,0 +1,391 @@
...
+        <tr>
+            <th>Method</th>
+            <td>
+                <code>fluid.textfieldStepper(container, options);</code>
+            </td>
+        </tr>
+        <tr>
+            <th>Description</th>
+            <td>
+                Instantiates the Textfield Stepper component. Connects a a textfield and a pair of increment/decrement buttons to set a value. Entries are restricted to numbers within a given range.
```
Comment (cindyli) [inline src/documents/TextfieldStepperAPI.md]: "Connects a textfield...". Remove the extra "a".

**Comment by cindyli** [comment]
Merged at 6c6bb9b49877a10a1b7979c77066d9e140c1c659

**Pull request merged**